src/lib.rs

```diff
@@ -39,8 +39,14 @@
 //! //std::thread::sleep(std::time::Duration::from_secs(4));
 //! ```
 
+/// Render quantum size as u32 (audio graph is rendered in blocks of this size)
+///
+/// derive BUFFER_SIZE from this const to prevent downcasting from `usize` to `u32`
+/// (cf. discussion [https://github.com/orottier/web-audio-api-rs/pull/52#issuecomment-969156219])
+pub const BUFFER_SIZE_U32: u32 = 128;
+
 /// Render quantum size (audio graph is rendered in blocks of this size)
-pub const BUFFER_SIZE: usize = 128;
+pub const BUFFER_SIZE: usize = BUFFER_SIZE_U32 as usize;
```

Comment:
My (not really opinionated) rationale here was that as `BUFFER_SIZE_U32` is finally only used in low-level `io.rs`, maybe it's good to reinforce that it is a kind of "special" `const` compared to `BUFFER_SIZE` (which should be preferred).

Reply:
I also prefer `BUFFER_SIZE` as a succinct name, conforming nicely to the spec naming.
If the U32 variant is only used in io.rs, let's move the const there so it will not pollute the pub namespace.
You could even use the usize variant in that file, and only cast at the final moment when passsing it to cpal. In that case we don't even need a const

Reply:
ok, used `u32::try_from(u: usize)` at last moment and clippy doesn't complain